File: app/src/main/java/me/tysheng/xishi/net/XishiService.java
package me.tysheng.xishi.net;

import io.reactivex.Flowable;
import me.tysheng.xishi.bean.DayAlbums;
import me.tysheng.xishi.bean.Mains;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * Created by Sty
 * Date: 16/8/22 22:18.
 */
public interface XishiService {


    @GET("mains/p{page}.html")
    Flowable<Mains> getMains(@Path("page") int page);

    @GET("albums/a{id}.html")
    Flowable<DayAlbums> getDayAlbums(@Path("id") int id);

}

File: app/src/main/java/me/tysheng/xishi/net/XishiService.kt
package me.tysheng.xishi.net

import io.reactivex.Observable
import me.tysheng.xishi.net.data.*
import retrofit2.http.Body
import retrofit2.http.POST

/**
 * Created by tysheng
 * Date: 21/9/18 6:21 PM.
 * Email: tyshengsx@gmail.com
 */
interface XishiService {

    @POST("albums/add")
    fun addAlbum(@Body album: ServerAlbum): Observable<CommonResponse<Any>>

    @POST("shots/add")
    fun addShot(@Body shot: Shot): Observable<CommonResponse<Any>>

    @POST("users/register")
    fun register(@Body registerParam: RegisterParam): Observable<CommonResponse<User>>

    companion object {
        const val token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VySWQiOjF9.KVG-MUZhzYWX0ycNNlu4t_jj77uWyB5PbpmB9StRZ3Q"
    }
}